docs/ota-client-guide/modules/ROOT/pages/sign-with-external-signature.adoc

```diff
@@ -0,0 +1,110 @@
+= Sign metadata with externally generated signature
+ifdef::env-github[]
+
+[NOTE]
+====
+We recommend that you link:https://docs.ota.here.com/ota-client/latest/{docname}.html[view this article in our documentation portal]. Not all of our articles render correctly in GitHub.
+====
+endif::[]
+
+For safety reasons you may want to sign your metadata with the signature generated by your own PKI.
+For more information on where to find your metadata files see the xref:finding-unsigned-metadata.adoc[Find the unsigned Root and Targets metadata].
+
+=== Signing Root metadata
+
+To sign `root.json` with your own generated signature you will need the name of key with which it was signed previously.
+You can use multiple signatures for signing.
+For verification purposes, with each signature provided you will need to specify the name of the key with which it was generated.
+
+```
+garage-sign root sign \
+        --repo <repository-name> \
+        --old-root-alias <key-name> \
+        --signatures <key-name=signature>
+```
+
+=== Signing Targets metadata
+
+For signing `targets.json` the procedure is similar except you don't need the old key name.
+
+NOTE: If you're passing a signature you cannot pass a version.
+
+```
+garage-sign targets sign \
+        --repo <repository-name> \
+        --signatures <key-name=signature>
+```
+
+== Example script with a flow that is using OpenSSL
+
+```
+#!/bin/bash
+
+set -x
+set -euo pipefail
+
+garage_sign="${1}"
+shift
+repo_name="${1}"
+shift
+online_credentials="${1}"
+shift
+root_pem_path="${1}"
+shift
+targets_pem_path="${1}"
+shift
+length="${1}"
+shift
+filename="${1}"
+shift
+version="${1}"
+shift
+sha="${1}"
+shift
+hardwareid="${1}"
+
+${garage_sign} init --repo ${repo_name} --credentials ${online_credentials}
+
+# generate keys to sign with
+openssl genpkey -algorithm RSA -out root_private.pem -pkeyopt rsa_keygen_bits:2048
+openssl rsa -pubout -in root_private.pem -out root_public.pem
+openssl genpkey -algorithm RSA -out targets_private.pem -pkeyopt rsa_keygen_bits:2048
+openssl rsa -pubout -in targets_private.pem -out targets_public.pem
+
+# initial pulling of unsigned metadata
+${garage_sign} targets pull --repo ${repo_name}
+${garage_sign} move-offline --repo ${repo_name} --old-root-alias origroot
+${garage_sign} root pull --repo ${repo_name}
+
+# import root public key
+echo '{"keyval":{"public":"'$(awk 1 ORS='\\n' < ${root_pem_path})'"},"keytype":"RSA"}' > tuf/${repo_name}/keys/root_public.pub
```

Comment:
I would use the new command `importpub` here and only publish this once garage-sign is updated.